src/test/shell/integration/target_compatible_with_test.sh

```diff
@@ -299,6 +299,59 @@ EOF
   expect_log 'Target //target_skipping:filegroup is incompatible and cannot be built'
 }
 
+# Validates that filegroups and other rules that don't inherit from
+# `NativeActionCreatingRule` can be marked with target_compatible_with. This is
```

Comment:
The `target_compatible_with` attribute is now on NativeBuildRule, not NativeActionCreatingRule.

And as soon as I submit #13223, filegroup will again be part of toolchain resolution (see the discussion on that PR about why).

My conclusion from #13223 is that there are no rules that are actual build rules and disable toolchain resolution, so I can't think of a rule to use in this test.

Reply:
I'm not really sure what you're looking for here.

Without this patch here, rules that don't inherit from `NativeActionCreatingRule` cannot use the `target_compatible_with` attribute. So this test will fail. With the patch, the desired functionality is implemented and the test starts passing.

Reply:
Sorry, I think I misunderstood (because I was still confused from the test I removed in #13223). I thought this test was relying on filegroup not doing toolchain resolution, but now I see that isn't the case. My bad!